src/main/java/Duke.java

```diff
@@ -1,55 +1,100 @@
 import java.util.Scanner;
 
 public class Duke {
-    public static void main(String[] args) {
-        String line = "____________________________________________________________";
-        System.out.println(line + "\n Hello! I'm Duke\n What can I do for you?\n" + line);
+
+    public static final String LINE = "____________________________________________________________";
+
+    public static void printList(Task[] list) {
+        System.out.println(LINE);
+        if (list[0] == null) {
+            System.out.println("No items added!");
+        }
+        int i = 0;
+        while (list[i] != null) {
+            System.out.println((i+1) + ". " + list[i]);
+            i += 1;
+        }
+        System.out.println(LINE);
+    }
+
+    public static void markTaskAsDone(Task[] list, int taskIndex) {
+        if (taskIndex < Task.numItemsAdded && taskIndex >= 0) {
+            if (list[taskIndex].isDone()) {
+                System.out.println(LINE + "\nThat task is already done!\n" + LINE);
+            } else {
+                list[taskIndex].markAsDone();
+                System.out.println(LINE + "\nNice! I've marked this task as done:");
+                System.out.println(list[taskIndex] + "\n" + LINE);
+            }
+        } else {
+            System.out.println("That task does not exist!\n" + LINE);
+        }
+    }
+
+    public static void printAddedTask(Task[] list) {
+        System.out.println(LINE + "\nGot it. I've added this task:");
+        System.out.println(list[Task.numItemsAdded-1]);
+        System.out.format("Now you have %d tasks in the list.\n" + LINE + "\n", Task.numItemsAdded);
+    }
+
+    public static void addTask(Task[] list, String input, String[] inputWords) {
+        String taskType = inputWords[0];
+        if (taskType.equalsIgnoreCase("todo")) {
+            String taskName = input.substring(5);
+            list[Task.numItemsAdded] = new Task(taskName);
+            Task.numItemsAdded += 1;
+            printAddedTask(list);
+        } else if (taskType.equalsIgnoreCase("deadline")) {
+            if (!input.contains("/by")) {
+                System.out.println(LINE + "\nIncorrect format for entering task!\n" + LINE);
+                return;
+            }
+            int taskEndIndex = input.indexOf("/by") - 1;
+            String taskName = input.substring(9, taskEndIndex);
+            String deadline = input.substring(taskEndIndex + 5);
+            list[Task.numItemsAdded] = new Deadline(taskName, deadline);
+            Task.numItemsAdded += 1;
+            printAddedTask(list);
+        } else if (taskType.equalsIgnoreCase("event")) {
+            if (!input.contains("/at")) {
+                System.out.println(LINE + "\nIncorrect format for entering task!\n" + LINE);
+                return;
+            }
+            int taskEndIndex = input.indexOf("/at") - 1;
+            String taskName = input.substring(6, taskEndIndex);
```

Comment:
repetitive work for taskName, deadline and printAddedTask etc, maybe can extract into a smaller functions